legend-pure-runtime/legend-pure-runtime-java-engine-compiled/src/main/java/org/finos/legend/pure/runtime/java/compiled/generation/processors/support/Pure.java

```diff
@@ -716,6 +718,33 @@ public static Object newObject(Bridge bridge, org.finos.legend.pure.m3.coreinsta
         {
             Class<?> c = ((CompiledExecutionSupport) es).getClassLoader().loadClass(JavaPackageAndImportBuilder.platformJavaPackage() + "." + Pure.elementToPath(aClass, "_", true) + "_Impl");
             Any result = (Any) c.getConstructor(String.class).newInstance(name);
+            // Set default values
+            RichIterable<CoreInstance> classProperties = _Class.getSimpleProperties(aClass, ((CompiledExecutionSupport) es).getProcessorSupport());
+            classProperties.forEach(new CheckedProcedure<CoreInstance>()
+            {
+                @Override
+                public void safeValue(CoreInstance p) throws Exception
+                {
+                    if (p instanceof Property<?, ?>)
+                    {
+                        Property<?, ?> prop = (Property<?, ?>) p;
+                        DefaultValue defaultValue = prop._defaultValue();
+                        if (defaultValue != null)
+                        {
+                            Object res = reactivate(defaultValue._functionDefinition()._expressionSequence().getFirst(), new PureMap(Maps.fixedSize.empty()), bridge, es);
+                            Method method = c.getMethod("_" + prop._name(), RichIterable.class);
+                            if (res instanceof RichIterable)
+                            {
+                                method.invoke(result, res);
+                            }
+                            else
+                            {
+                                method.invoke(result, Lists.fixedSize.of(res));
+                            }
+                        }
+                    }
+                }
+            });
```

Comment:
I think this block should go after the key expressions, and we should only invoke the default value if the value isn't set by a key expression. That means we'd need to keep track of which properties we've set from key expressions, but that's easy enough.

Also, I think we should just cast to Property<?, ?> here and not bother with the if check. After all, if it's not an instance of Property, we want to throw instead of just ignoring it.